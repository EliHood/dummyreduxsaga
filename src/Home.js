import React, { Component } from 'react';
import PropTypes from 'prop-types'; 
class Home extends Component{
    static defaultProps = {
       fetchDataInit: () => {}
    }
    state = {
        hide:true,
        data:[],
        stateData:[]
    }
    componentDidMount(){
        this.props.fetchDataInit()
    }
    componentWillUpdate(prevProps){
        if(this.props.data.data !== prevProps.data.data){
            this.setState({
                hide:false,
            })      
        }
   }
    render(){
        const {hide} = this.state
        if(hide){
            return(
                <h3>Loading...</h3>
            )
        }
        return(
           <div>
                <h1>Welcome Home</h1>
                {this.props.data.data.map( (item, i) => (
                    <div key={i}> 
                        <h4>{item.title}</h4>
                        <small>{item.completed.toString()}</small>
                    </div>
                ))}
           </div>
        )
    }
}

export default Home