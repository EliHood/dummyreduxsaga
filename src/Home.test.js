import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Home from './Home';

describe('Should render <Home/> Component', () => {

  it('should match <Home/> snapshot', () => {
    const wrapper = shallow(<Home />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should find h3', () => {
    const wrapper = shallow(<Home />);
    expect(wrapper.find('h3').text()).toBe('Loading...');
  });
});
