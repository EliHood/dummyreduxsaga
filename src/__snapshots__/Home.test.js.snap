// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render <Home/> Component should match <Home/> snapshot 1`] = `
<h3>
  Loading...
</h3>
`;
